db/migrate/20250214192930_add_unique_index_email_entreprise_to_customers.rb

```diff
@@ -0,0 +1,8 @@
+# frozen_string_literal: true
+
+class AddUniqueIndexEmailEntrepriseToCustomers < ActiveRecord::Migration[7.0]
+  def change
+    remove_index :customers, :email, name: :index_customers_on_email
+    add_index(:customers, [:email, :enterprise_id], unique: true, name: :index_customers_on_email)
```

Comment:
I was wondering why you removed the previous index instead of adding a new one. Apparently, when you have a multi-column index, you can also use it for the first column of that index alone. So queries by `email` alone are covered by the new index as well. :+1:

Reply:
I have added the run migration in the "What should we test" bullet.